src/stactools/package/stac.py

```diff
@@ -0,0 +1,179 @@
+from datetime import datetime
+import logging
+
+from shapely.geometry import Polygon
+from pystac.extensions.projection import ProjectionExtension
+from pystac import (
+    CatalogType,
+    Collection,
+    Item,
+    Asset,
+    Extent,
+    SpatialExtent,
+    TemporalExtent,
+    Link,
+    Provider,
+    MediaType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# This includes a set of minimum requirements for the creation of a STAC Collection,
+# and should be added to/modified as necessary to develop a dataset-specific package.
+def create_collection(
+    title: str,
+    description: str,
+    catalog_type: str,
+    providers: list[Provider],
+    license: str,
+    license_link: Link,
+    extent: list,
+    start_time: datetime,
+    end_time: datetime,
+    href: str = None,
+) -> Collection:
+    """Create a STAC Collection
+
+    Args:
+        title (str): Collection title
+        description (str): Collection description
+        catalog_type (str): One of
+        'ABSOLUTE_PUBLISHED', 'RELATIVE_PUBLISHED', 'SELF_CONTAINED'
+        providers (list[Provider]): One or more providers,
+        see https://pystac.readthedocs.io/en/latest/api.html#provider
+        license (str): Data provider license token
+        license_link (Link): license Link object,
+        see https://pystac.readthedocs.io/en/latest/api.html#link
+        extent (list): Extent. ex [xmin, ymin, xmax, ymax].
+        See https://pystac.readthedocs.io/en/latest/api.html#spatialextent
+        start_time (datetime): Collection start time object- must be in UTC time zone
+        end_time (datetime): Collection end time object- must be in UTC time zone
+        href (str, optional): Location to store Collection json. Defaults to None,
+        and no json will be created.
+
+    Returns:
+        Collection: STAC Collection object
+    """
+    # The ID may be changed to be provided as an arg if desired
+    id = title.replace(" ", "-")
+
+    collection = Collection(
+        id=id,
+        title=title,
+        description=description,
+        providers=providers,
+        license=license,
+        extent=Extent(SpatialExtent([extent]),
+                      TemporalExtent([start_time, end_time])),
+        catalog_type=getattr(CatalogType, catalog_type),
+    )
+    collection.add_link(license_link)
+
+    if href:
+        collection.set_self_href(href)
+        collection.save_object()
```

Comment:
I'm not sure if a function named `create_collection` should also save that collection sometimes, but not always (Single Responsibility Principle).